### PR TITLE
Fill out a changelog for the 1.1.0 release

### PR DESCRIPTION
**Pull request by huonw**
This fills out the changelog for development since 1.0.0, in preparation for the 1.1.0 release.

This has covered: 

- https://github.com/stellargraph/stellargraph/compare/v1.0.0...e3bf444
- https://github.com/stellargraph/stellargraph/compare/e3bf444...7234813

Waiting on:

- [x] #1312/#1650
- [x] #1534/#1632
- [x] #1592/#1598
- [x] #1635/#1636
- [x] #1637/#1651
- [x] #1638/#1643
- [x] #1639/#1645
- [x] #1640/#1652
- [x] #1642/#1644
- [x] #1646/#1649
- [x] #1648/#1653

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,89 @@
...
+[unsup-rgcn-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/demos/embeddings/deep-graph-infomax-embeddings.html#Heteogeneous-models
+[keras-n2v-rl-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/demos/embeddings/keras-node2vec-embeddings.html
+[keras-n2v-nc-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/demos/node-classification/keras-node2vec-node-classification.html
+[a2v-rl-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/demos/embeddings/attri2vec-embeddings.html
+[a2v-nc-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/demos/node-classification/attri2vec-node-classification.html
+[a2v-lp-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/demos/link-prediction/attri2vec-link-prediction.html
+
+### Major features and improvements
+
+- Support for the Neo4j graph database has been much improved:
```
Comment (timpitman) [inline CHANGELOG.md:35]: possibly a bit colloquial?  simple fix below, but might be an even better wording?
```suggestion
- Support for the Neo4j graph database has been significantly improved:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,89 @@
...
+  - The peak memory usage of construction and adjacency list building is now monitored to ensure that there are not large spikes for large graphs, that exceed available memory [\#1546](https://github.com/stellargraph/stellargraph/pull/1546). This peak usage has thus been optimised: [\#1551](https://github.com/stellargraph/stellargraph/pull/1551),
+  - Other optimisations: the `edge_arrays`, `neighbor_arrays`, `in_node_arrays` and `out_node_arrays` methods have been added, reducing time and memory overhead by leaving data as its underlying NumPy array [\#1253](https://github.com/stellargraph/stellargraph/pull/1253); the `node_type` method now supports multiple nodes as input, making algorithms like HinSAGE and Metapath2Vec much faster [\#1452](https://github.com/stellargraph/stellargraph/pull/1452); the default edge weight of 1 no longer consumes significant memory [\#1610](https://github.com/stellargraph/stellargraph/pull/1610).
+- Overall performance and memory usage improvements since 1.0.0, in numbers:
```
Comment (timpitman) [inline CHANGELOG.md:52]: 👏

**Comment by huonw** [comment]
I've now included everything up to 7234813, and preemptively #1652.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,96 @@
...
+    - construction without node features is now 2.3× faster, uses 31% less memory and has a memory peak 57% smaller.
+    - construction with node features from NumPy arrays is 6.8× faster, uses 6.5% less memory overall and 85% less new memory (the majority of the memory is shared with the original NumPy arrays), and has a memory peak (above the raw data set) 70% smaller, compared to Pandas DataFrames in 1.0.0.
+    - adjacency lists are 4.7-5.0× faster to construct, use 28% less memory and have a memory peak 60% smaller.
+  - Various random walkers are faster: `BiasedRandomWalk` is up to 30× faster with weights and 5× faster without weights on MovieLens and up to 100× faster on some synthetic datasets, `UniformRandomMetapathWalk` is up to 17× faster (on MovieLens), `UniformRandomWalk` is up to 1.4× (on MovieLens).
+- Tensorflow 2.2 and thus Python 3.8 are now supported [\#1278](https://github.com/stellargraph/stellargraph/pull/1278)
+
+[glossary-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/glossary.html
+[neo4j-sg-1.1.0]: https://stellargraph.readthedocs.io/en/latest/api.html#stellargraph.connector.neo4j.Neo4jStellarGraph
```
Comment (kjun9) [inline CHANGELOG.md:62]: This should probably point to v1.1.0?
```suggestion
[neo4j-sg-1.1.0]: https://stellargraph.readthedocs.io/en/v1.1.0/api.html#stellargraph.connector.neo4j.Neo4jStellarGraph
```

**Comment by huonw** [comment]
Added #1653 preemptively too, since that does seem to be working as intended, would probably make our life slightly easier for the release, and so I'll wait for it to merge before cutting the release.

**Comment by huonw** [comment]
The CI failures were spurious (#1115), and this PR is just touching a markdown file, so "force merged".

**Pull request merged**